src/Modal/index.jsx

```diff
@@ -168,7 +167,22 @@ class Modal extends React.Component {
         aria-modal
         aria-labelledby={this.headerId}
       >
-        <div className={styles['modal-dialog']}>
+        <div
+          className={classNames({ [styles['modal-backdrop']]: open })}
+          role="presentation"
+          onClick={this.close}
+        />
+        <div
+          className={classNames(
+            styles.modal,
+            styles['modal-dialog'],
+            {
+              [styles['modal-open']]: open,
+              [styles.show]: open,
+              [styles.fade]: !open,
```

Comment:
It looks good in Storybook, but now we have the `modal-open`, `show`, and `fade` styles at two levels of the Modal tree?

Reply:
I know, it's not my favourite. The main benefit of this approach is that it simplifies the close click handler logic considerably. I will try taking another stab at it without updating the DOM, and just using move logic in the click handler.

Reply:
Ah alright, I was mostly asking to understand if that had been left in there on purpose or not. I'll keep an eye out for that next commit before trying to validate further then. 👍